tests/ui/traits/pointee-normalize-equate.rs

```diff
@@ -0,0 +1,41 @@
+// check-pass
+
```

Comment:
```suggestion
// revisions: next old
//[next] compile-flags: -Znext-solver
```
I can help if the next solver does not already support this

Reply:
The new solver does indeed not support this:
```
error[E0271]: type mismatch resolving `<T as Pointee>::Metadata == <Wrapper<T> as Pointee>::Metadata`
  --> src/lib.rs:21:20
   |
21 |     cast_same_meta(ptr)
   |     -------------- ^^^ types differ
   |     |
   |     required by a bound introduced by this call
   |
```

The relevant code seems to be here and it looks like it's already supposed to normalize from `Wrapper<T>::Metadata` to `T::Metadata`:

https://github.com/rust-lang/rust/blob/af08c64e3800fd15f0afdb8651de8fa7dacc0026/compiler/rustc_trait_selection/src/solve/normalizes_to/mod.rs#L410-L435

I don't know anything about the new solver, so any help to get this working would be appreciated.

Reply:
Update: I've pushed a commit that appears to make it work with the new solver, but I don't know what I'm doing so please review carefully.